### ISSUE TITLE
GCP BigQuery Data Transfer Run Issue

### ISSUE THREAD

**Issue by totogo** [apache/airflow#15088]
**Apache Airflow version**: composer-1.15.1-airflow-1.10.14

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`):

**Environment**:

- **Cloud provider or hardware configuration**: Google Composer 

**What happened**:

After a `TransferConfig` is created successfully by operator `BigQueryCreateDataTransferOperator`, and also confirmed in GCP console, the created Resource name is:
```
projects/<project-id>/locations/europe/transferConfigs/<transfer-config-id>
```
Then I use operator `BigQueryDataTransferServiceStartTransferRunsOperator` to run the transfer, I got this error:
```
Traceback (most recent call last)
  File "/usr/local/lib/airflow/airflow/models/taskinstance.py", line 985, in _run_raw_tas
    result = task_copy.execute(context=context
  File "/usr/local/lib/airflow/airflow/providers/google/cloud/operators/bigquery_dts.py", line 290, in execut
    metadata=self.metadata
  File "/usr/local/lib/airflow/airflow/providers/google/common/hooks/base_google.py", line 425, in inner_wrappe
    return func(self, *args, **kwargs
  File "/usr/local/lib/airflow/airflow/providers/google/cloud/hooks/bigquery_dts.py", line 235, in start_manual_transfer_run
    metadata=metadata or ()
  File "/opt/python3.6/lib/python3.6/site-packages/google/cloud/bigquery_datatransfer_v1/services/data_transfer_service/client.py", line 1110, in start_manual_transfer_run
    response = rpc(request, retry=retry, timeout=timeout, metadata=metadata,
  File "/opt/python3.6/lib/python3.6/site-packages/google/api_core/gapic_v1/method.py", line 145, in __call_
    return wrapped_func(*args, **kwargs
  File "/opt/python3.6/lib/python3.6/site-packages/google/api_core/grpc_helpers.py", line 75, in error_remapped_callabl
    six.raise_from(exceptions.from_grpc_error(exc), exc
  File "<string>", line 3, in raise_fro
google.api_core.exceptions.NotFound: 404 Requested entity was not found
```

**What you expected to happen**:

`BigQueryDataTransferServiceStartTransferRunsOperator` should run the data transfer job. 

**How to reproduce it**:

1. Create a `cross_region_copy` TransferConfig with operator`BigQueryCreateDataTransferOperator`
2. Run the job with operator `BigQueryDataTransferServiceStartTransferRunsOperator`
```python
create_transfer = BigQueryCreateDataTransferOperator(
    task_id=f'create_{ds}_transfer',
    transfer_config={
        'destination_dataset_id': ds,
        'display_name': f'Copy {ds}',
        'data_source_id': 'cross_region_copy',
        'schedule_options': {'disable_auto_scheduling': True},
        'params': {
            'source_project_id': source_project,
            'source_dataset_id': ds,
            'overwrite_destination_table': True
        },
    },
    project_id=target_project,
)
transfer_config_id = f"{{{{ task_instance.xcom_pull('create_{ds}_transfer', key='transfer_config_id') }}}}"

start_transfer = BigQueryDataTransferServiceStartTransferRunsOperator(
    task_id=f'start_{ds}_transfer',
    transfer_config_id=transfer_config_id,
    requested_run_time={"seconds": int(time.time() + 60)},
    project_id=target_project,
)
run_id = f"{{{{ task_instance.xcom_pull('start_{ds}_transfer', key='run_id') }}}}"
```

**Anything else we need to know**:

So I went to the [Google's API reference page](https://cloud.google.com/bigquery-transfer/docs/reference/datatransfer/rest/v1/projects.locations.transferConfigs/startManualRuns) to run some test. When I use this parent parameter `projects/{projectId}/transferConfigs/{configId}`, it thrown the same error. But it works when I use `rojects/{projectId}/locations/{locationId}/transferConfigs/{configId}`

I guess the piece of code that causes this issue is here in the hook, why does it use `projects/{projectId}/transferConfigs/{configId}` instead of `rojects/{projectId}/locations/{locationId}/transferConfigs/{configId}`?

https://github.com/apache/airflow/blob/def961512904443db90e0a980c43dc4d8f8328d5/airflow/providers/google/cloud/hooks/bigquery_dts.py#L226-L232

**Comment by mik-laj** [comment]
> I guess the piece of code that causes this issue is here in the hook, why does it use projects/{projectId}/transferConfigs/{configId} instead of rojects/{projectId}/locations/{locationId}/transferConfigs/{configId}?

When this operator was created, Google only supported one location and then made a breaking change to the API to support multiple locations.

I think PR is necessary here to update to the new API.

**Comment by totogo** [comment]
> When this operator was created, Google only supported one location and then made a breaking change to the API to support multiple locations.
> 
> I think PR is necessary here to update to the new API.

@mik-laj I see, right now I pass the location as part of `project_id` in the operator as a workaround, i.e. change `project_id=<project-id>` to `project_id=<project-id>/locations/<location>`, which works. 

I can work on the PR. The proper solution should be to add a new argument `location_id` in the hooks and operators? or to add the full resouce name? Since it is already returned by the operator `BigQueryCreateDataTransferOperator` in Xcom `return_value`:
```
{'name': 'projects/<project-id>/locations/europe/transferConfigs/<transfer-config-id>', 
'destination_dataset_id': 'dest', 
'display_name': 'Copy source to dest', 
'update_time': '2021-03-31T07:31:54.655068Z', 
'data_source_id': 'cross_region_copy',
'params': {'overwrite_destination_table': True, 'source_dataset_id': 'source', 'source_project_id': '<source-project-id>'}, 'user_id': 'xxxx', 
'dataset_region': 'europe', 
'schedule_options': {'disable_auto_scheduling': True}, 
'schedule': '', 'data_refresh_window_days': 0,
'disabled': False, 
'state': 0, 
'notification_pubsub_topic': ''}
```

**Comment by mik-laj** [comment]
> I can work on the PR. The proper solution should be to add a new argument location_id in the hooks and operators?

New argument.  See: https://docs.google.com/document/d/1_rTdJSLCt0eyrAylmmgYc3yZr-_h51fVlnvMmWqhCkY/edit?ts=5bb72dfd#heading=h.3sxddzjr5f6q Section: Working with parent / name variables

**Comment by totogo** [comment]
> New argument. See: https://docs.google.com/document/d/1_rTdJSLCt0eyrAylmmgYc3yZr-_h51fVlnvMmWqhCkY/edit?ts=5bb72dfd#heading=h.3sxddzjr5f6q Section: Working with parent / name variables

@mik-laj sorry this document is not publicly accessible?

**Comment by mik-laj** [comment]
![Screenshot_20210407-164724.jpg](https://user-images.githubusercontent.com/12058428/113886425-f41d2d80-97c0-11eb-9ca0-04b574fb6832.jpg)

**Comment by ashishpatel0720** [comment]
@totogo are you working on it,
if not, I can start working on it.

**Comment by pateash** [comment]
@keweishang Hi, I haven't started looking into it as.
@totogo is working on it.

**Comment by bradleybonitatibus** [comment]
My team is running into issues with this operator in Airflow 2+. Is this open for contribution?

**Comment by pateash** [comment]
@eladkal Sorry, I didn't get a chance to look into it.

@bradleybonitatibus Yes, it is open for contribution.
please feel free to start working on it.

**Comment by bradleybonitatibus** [comment]
Resolved by #20221
